registries/adapters/adapter.go

```diff
@@ -57,21 +58,32 @@ type Configuration struct {
 	Tag        string
 }
 
-// Retrieve the spec from a registry manifest request
-func imageToSpec(req *http.Request, image string) (*bundle.Spec, error) {
-	log.Debug("Registry::imageToSpec")
-	spec := &bundle.Spec{}
-	req.Header.Add("Accept", "application/json")
-
-	resp, err := http.DefaultClient.Do(req)
+func commonResponseHandler(resp *http.Response) ([]byte, error) {
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		return nil, err
 	}
-	defer resp.Body.Close()
+	if resp.StatusCode == http.StatusUnauthorized {
+		log.Errorf("Unable to authenticate to the registry, registry credentials could be invalid.")
+		return nil, nil
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		log.Errorf("unexpected response code %v body %v", resp.StatusCode, string(body))
+		return nil, nil
+	}
+	return body, nil
+}
 
+// Retrieve the spec from a registry manifest request
+func imageToSpec(imageDetails []byte, image string) (*bundle.Spec, error) {
```

Comment:
This makes way more sense. The method simply converts data not how to get the data. :+1: